README.md

```diff
@@ -1,2 +1,93 @@
-# workflow-migration
-A simple k8s job to migrate workflow from helm-classic to helm
+# Workflow Migration
+
+Deis (pronounced DAY-iss) Workflow is an open source Platform as a Service (PaaS) that adds a developer-friendly layer to any [Kubernetes](http://kubernetes.io) cluster, making it easy to deploy and manage applications on your own servers.
+
+For more information about the Deis Workflow, please visit the main project page at https://github.com/deis/workflow.
+
+We welcome your input! If you have feedback, please [submit an issue][issues]. If you'd like to participate in development, please read the "Development" section below and [submit a pull request][prs].
+
+# About
+The Workflow Migration service is used to migrate from an [helm-classic](https://github.com/helm/helm-classic) install of the workflow to [kubernetes helm](https://github.com/kubernetes/helm) without destroying the existing cluster or having any downtime for the apps. It does so by first checking the current install of workflow and creating a release artifact similar to the one kubernetes helm creates during an install thereby making kubernetes helm think that the current install is actually created by it. Then workflow can be simply upgraded whenever needed using the kubernetes helm charts.
+
+> Warning: Only workflow install on or after v2.6.0 can be upgraded using this migration service.
+
+# Usage
+1) Check that kubernetes helm and its corresponding server component tiller are [installed](https://github.com/kubernetes/helm/blob/master/docs/install.md).
+
+2) Fetch the builder and controller deployment objects just to make sure that the existing install state can achieved if the deis migration service fails. Deis migration service deletes the builder and controller deployment objects because of an [issue](https://github.com/kubernetes/kubernetes/issues/35134) in kubernetes with the patching.
+```
+$ kubectl --namespace=deis get deployment deis-builder -o yaml > ~/active-deis-builder-deployment.yaml
+$ kubectl --namespace=deis get deployment deis-controller -o yaml > ~/active-deis-controller-deployment.yaml
+```
+
+3) Run the migration service to create a helm release object based on the current workflow install. If not otherwise specified, the workflow_release_name will be `deis-workflow` and workflow_version will be `v2.7.0`.
+```
+$ git clone https://github.com/deis/workflow-migration.git
+$ cd workflow-migration
+$ helm install ./charts/workflow-migration/ --set workflow_release_name=<optional release name for the helm>,workflow_version=<optional current version of workflow>
+```
+or
+```
+$ helm repo add deismigration https://charts.deis.com/migration
```

Comment:
the chart repo will probably be named `workflow-migration` as opposed to just `migration`, correct?